lib/chef/cookbook/metadata.rb

```diff
@@ -386,6 +397,28 @@ def replaces(cookbook, *version_args)
         @replacing[cookbook]
       end
 
+      # Metadata DSL to set a valid chef_version.  May be declared multiple times
+      # with the result being 'OR'd such that if any statements match, the version
+      # is considered supported.  Uses Gem::Requirement for its implementation.
+      #
+      # @param version_args [Array<String>] Version constraint in String form
+      # @return [Array<Gem::Dependency>] Current chef_versions array
+      def chef_version(*version_args)
+        @chef_versions << Gem::Dependency.new('chef', *version_args)
```

Comment:
The fact that this returns a value makes it sound like you expect it to be used as a reader too? Calling `chef_versions()` would shift on an empty dependency, maybe add `unless version_args.empty?` here?